app/controllers/samples_controller.rb

```diff
@@ -358,7 +358,7 @@ def results_folder
   end
 
   def fastqs_folder
-    @file_list = @sample.fastqs_folder_files
+    @file_list = current_power.host_viewable_samples.include?(@sample) ? @sample.fastqs_folder_files : []
```

Comment:
I think you can just do current_power.updatable_samples.find_by(id: @sample.id)  or just use current_power.updatable_samples